### ISSUE TITLE
Client for Nodejs

### ISSUE THREAD

**Issue by itaditya** [internetarchive/openlibrary-client#38]
The Open Library client mentioned in the README is an angular module. This has two major disadvantages - 
1. Developers can't use it with other frameworks like React or Vue. They can't even use it with Vanilla Js or jQuery.
2. No cli support. To use client on the command line, devs have to use the python client. For those developers using Nodejs as their backend, this can be a deal breaker.

It would be much better if a client is made on Node.js first which can provide the option to usde the tool on the command line as well as their Nodejs script. 
I would love to make this client

**Comment by mekarpeles** [comment]
Your effort is welcome! 

Consider though that the majority of the things open library client does are administrative (used for scripting bots which update data) and these actions are unlikely to be performed within a webpage. Most API endpoints don't require any authentication and are already fairly straight forward to call in react etc. If there are specific features you want to surface which require authentication, I'd focus on on making a vanilla is lib which makes those easy. I think creating a command line version in node may even be value subtracting if it different from the reference implementation. It adds overhead for us to check and risks users making inconsistent metadata changes in open library.

A vanilla js library sound like a useful idea. Your obviously welcome to do whatever you prefer; these are my preferences

**Comment by itaditya** [comment]
You are correct. I can think only one advantage of making a npm module which can be used with node. That is many times people need to access the data from their backend, rather than frontend. 

Reason - As backend is much closer to their db, they can easily fetch the data and store in their db which can be later utilized by them for their own application.

**Comment by itaditya** [comment]
Also, I'll try to make the plugin in such a way that it can be used both as client side lib or backend lib like lodash and other popular libraries does.

**Comment by mekarpeles** [comment]
Sure! Looking forward to hearing about your progress. If you have questions along the way, feel free to ping us on our gitter

**Comment by tfmorris** [comment]
My suggestion was shot down for the Python library, but I'd like to suggest that you use a permissive license for your new Javascript library so that all can use it freely.

**Comment by itaditya** [comment]
ok

**Comment by mekarpeles** [comment]
@itaditya any questions so far / anything we can take a look at? :)

**Comment by itaditya** [comment]
@mekarpeles , I was really busy in my exams. I'll be working on this in few days.
